### PR TITLE
Centralized CA setup changes

### PR DESCRIPTION
**Pull request by Garima-msft**
## Description
</br>

1. PATToken URL support for Centralized CA

 - ICA new param : PATTokenURL, IdentityResourceId
 - UCA new param : PATTokenURL.
 
2. Webhook settings
- enable webhook setting from UCA
- UCA new param: webhookurl, webhookauthzheadername, webhookauthzheadervalue, AllowSelfSignedWebhookCertificate

3. If projectname * is used, resolve it to comma seperated project names in ICA,UCA

 
## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/ContinuousAssurance/CAAutomation.ps1 [hunk 1]
```diff
@@ -317,7 +331,24 @@ class CAAutomation : ADOSVTCommandBase
 						$output = 'OK'
 					}
 				}
-			}
+            }
+            #Resolve projectNames if * is used in command
+            if ($this.ProjectNames -eq "*")
+            {
+                $apiURL = 'https://dev.azure.com/{0}/_apis/projects?$top=500&api-version=5.1' -f $($this.SubscriptionContext.SubscriptionName);
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Core/ContinuousAssurance/CAAutomation.ps1:338]: api change as per pagination logic

File: src/AzSK.ADO/Framework/Core/ContinuousAssurance/CAAutomation.ps1 [hunk 2]
```diff
@@ -227,7 +241,7 @@ class CAAutomation : ADOSVTCommandBase
 				{
 					$this.updateSecret = $true
 				}
-				if($_.Key -eq "ClearExtendedCommand")
+				if($_.Key -eq "ClearExtendedCommand" -or $_.Key -ne "PATTokenURL")
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Core/ContinuousAssurance/CAAutomation.ps1:244]: do we need this?

File: src/AzSK.ADO/Framework/Core/ContinuousAssurance/CAAutomation.ps1 (hunk 1 above)
Comment (Garima-msft) [reply src/AzSK.ADO/Framework/Core/ContinuousAssurance/CAAutomation.ps1]: Done

File: src/AzSK.ADO/Framework/Core/ContinuousAssurance/CAAutomation.ps1 (hunk 2 above)
Comment (Garima-msft) [reply src/AzSK.ADO/Framework/Core/ContinuousAssurance/CAAutomation.ps1]: removed this block of code

**Review by abhaydaga — APPROVED** [review]

**Pull request merged**